cast/src/lib.rs

```diff
@@ -1077,6 +1080,85 @@ impl SimpleCast {
         }?)
     }
 
+    /// Encodes a string or list to hexadecimal rlp
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///    assert_eq!(Cast::to_rlp("\"foundry\"".to_string()).unwrap(), "0x87666f756e647279" );
+    ///    assert_eq!(Cast::to_rlp("[]".to_string()).unwrap(), "0xc0");
+    ///    assert_eq!(Cast::to_rlp("[\"a\"]".to_string()).unwrap(), "0xc161");
+    ///    assert_eq!(Cast::to_rlp("[\"a\",\"b\"]".to_string()).unwrap(), "0xc26162");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn to_rlp(value: String) -> Result<String> {
+        let val = serde_json::from_str(&value)?;
+        let item = rlp_converter::value_to_item(&val, false);
+        Ok(format!("0x{}", hex::encode(rlp::encode(&item))))
+    }
+
+    /// Decodes rlp encoded hexadecimal string
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::from_rlp("0x87666f756e647279".to_string()).unwrap(), "\"foundry\"");
+    ///     assert_eq!(Cast::from_rlp("87666f756e647279".to_string()).unwrap(), "\"foundry\"");
+    ///     assert_eq!(Cast::from_rlp("0xc0".to_string()).unwrap(), "[]");
+    ///     assert_eq!(Cast::from_rlp("c161".to_string()).unwrap(), "[\"a\"]");
+    ///     assert_eq!(Cast::from_rlp("0xc26162".to_string()).unwrap(), "[\"a\",\"b\"]");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn from_rlp(value: String) -> Result<String> {
+        let striped_value = strip_0x(&value);
+        let bytes = hex::decode(striped_value).expect("could not decode hex");
+        Ok(format!("{}", rlp::decode::<Item>(&bytes).expect("rlp decoding failed")))
+    }
+
+    /// Encodes hex data or list of hex data to hexadecimal rlp
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::hex_to_rlp("[]").unwrap(),"0xc0".to_string());
+    ///     assert_eq!(Cast::hex_to_rlp("0x22").unwrap(),"0x22".to_string());
+    ///     assert_eq!(Cast::hex_to_rlp("[\"0x61\"]",).unwrap(), "0xc161".to_string());
+    ///     assert_eq!(Cast::hex_to_rlp( "[\"0x61\",\"62\"]").unwrap(), "0xc26162".to_string());
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn hex_to_rlp(value: &str) -> Result<String> {
```

Comment:
Is this a superset of `to_rlp`? If so, is there any reason we should have both?

Reply:
As suggested in the issue it allows to chose between passing string and hex in the input array, in the example below the `hex_to_rlp` directly passes the 4 bytes to rlp (0x prefix is optional) while `to_rlp` example encode the "5555666" string

> ![Screenshot from 2022-05-25 23-52-47](https://user-images.githubusercontent.com/104269768/170373962-348d7932-b54e-4e5a-a2d3-0380864f029f.png)

Just like for `from_rlp` i'm not sure about the use cases of the string version, i guess the user can hex encode a string before passing to `to_rlp`

Reply:
I'm still not entirely sure I understand the distinction, what bytes is it passing in `--hex-to-rlp`?

Reply:
here is a better example to clarify, all these commands have the same output:

```
cast --to-rlp '"UfUf"'
0x8455665566
cast --hex-to-rlp '"0x55665566"'
0x8455665566
cast --hex-to-rlp '"55665566"'
0x8455665566
```

`--to-rlp` converts the string to bytes before encoding while `--hex-to-rlp` expects input values to be hex encoded.

I suggest renaming the hex version `--to-rlp` and the other one `--str-to-rlp` if we are to keep it.
That way both `--to-rlp` and `--from-rlp` use hex encoded data.